mcodingbot/plugins/python.py

```diff
@@ -0,0 +1,51 @@
+import re
+
+import crescent
+import hikari
+
+from mcodingbot.utils import Plugin
+
+plugin = Plugin()
+
+pep_regex = re.compile(r"pep *\d{1,4}", re.IGNORECASE)
+
+
+def extract_pep_number(ref: str) -> int:
+    return int(ref.casefold()[3:].strip())
```

Comment:
Why casefold?

Reply:
Because we need to make the string case insensitive.

Reply:
numbers don't have a case

Reply:
it's about "PEP", that we remove before converting the string to an int.

Reply:
The length of PEP doesn't change based on whether or not it's capitalized. Also, since the regex specifies an arbitrary number of spaces, this will break if there are 2 or 0 spaces.

I think it would be better to use a named capture group for the number part of the regex and just grab that explicit (and remove this particular function).

Reply:
> The length of PEP doesn't change based on whether or not it's capitalized
Well, that's because it was using `removeprefix()` before, but nox complained about it so I changed and I forgot that `casefold()` was unnecessary